internal/consensus/replay.go

```diff
@@ -304,6 +304,10 @@ func (h *Handshaker) ReplayBlocks(
 	if appBlockHeight == 0 {
 		validators := make([]*types.Validator, len(h.genDoc.Validators))
 		for i, val := range h.genDoc.Validators {
+			// Ensure that the public key type is supported.
```

Comment:
Why wasn't this logic here before, was it a bug, or was is elsewhere?

Reply:
Before, we're panicking here if the key type is unknown.